candle-kernels/src/cast.cu

```diff
@@ -13,13 +13,13 @@ extern "C" __global__ void FN_NAME( \
     const size_t *strides = info + num_dims; \
     if (is_contiguous(num_dims, dims, strides)) { \
         for (unsigned int i = blockIdx.x * blockDim.x + threadIdx.x; i < numel; i += blockDim.x * gridDim.x) { \
-            out[i] = inp[i]; \
+            out[i] = DST_TYPENAME(inp[i]); \
```

Comment:
If this version works that's fine but if it doesn't create issue maybe a `static_cast` would be the most idiomatic way?

Reply:
(CI actually doesn't seem very happy with this "function call" version)

Reply:
Quite odd, I can't reproduce on 11.8 on the host machine, even when forcing CUDA_11.8.

Reply:
Ok reproduced, the issue is not in the 11.8 headers, but in the 11.8 nvcc.

Looking how to fix.

Reply:
Ok the `bf16 <-> f16` cast is actually missing in cu118....

Reply:
I don't feel good at all with the proposed fix... There must be a better way..